src/main/java/moonset/metastore/sync/MetastoreSyncUtils.java

```diff
@@ -0,0 +1,200 @@
+package moonset.metastore.sync;
+
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.collections.MapUtils;
+import org.apache.hadoop.hive.metastore.IMetaStoreClient;
+import org.apache.hadoop.hive.metastore.api.Database;
+import org.apache.hadoop.hive.metastore.api.NoSuchObjectException;
+import org.apache.hadoop.hive.metastore.api.Partition;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.thrift.TException;
+
+import moonset.metastore.sync.catalog.CatalogClient;
+import moonset.metastore.sync.exception.MetastoreException;
+import moonset.metastore.sync.util.PartitionUtils;
+import moonset.metastore.sync.util.TableUtils;
+import com.google.common.collect.Lists;
+
+import lombok.extern.slf4j.Slf4j;
+
+/** The utility class for metastore sync process. */
+@Slf4j
+public final class MetastoreSyncUtils {
+
+    /** Prevent the class to be instanced. */
+    private MetastoreSyncUtils() {}
+
+    /** Batch threshold for add_partitions() method. */
+    private static final int BATCH_SIZE = 30;
+
+    /** Sync all partitions of the target name from source metastore to dest metastore. */
+    public static void syncAllPartitions(
+            final IMetaStoreClient source,
+            final IMetaStoreClient dest,
+            final String srcDatabaseName,
+            final String srcTableName,
+            final String destDatabaseName,
+            final String destTableName)
+            throws MetastoreException {
+        CatalogClient client = new CatalogClient(source);
+        syncPartitions(dest, destDatabaseName, destTableName, client.getAllPartitions(srcDatabaseName, srcTableName));
+    }
+
+    /**
+     * Sync partitions which match the patterns in <code>partVals</code> of the target table from
+     * source metastore to dest metastore.
+     */
+    public static void syncPartitions(
+            final IMetaStoreClient source,
+            final IMetaStoreClient dest,
+            final String srcDatabaseName,
+            final String srcTableName,
+            final String destDatabaseName,
+            final String destTableName,
+            final Map<String, String> partVals)
+            throws MetastoreException {
+        List<Partition> partitions = null;
+        // We don't use pagination query since partition size is not large enough, and we will
+        // need not return partition to client
+        if (MapUtils.isEmpty(partVals)) {
+            partitions = Lists.newArrayList();
+        } else {
+            // Only load specify partitions
+            CatalogClient client = new CatalogClient(source);
+            partitions = client.getPartitions(srcDatabaseName, srcTableName, partVals);
+        }
```

Comment:
I prefer the style like this to show clearer logic to read:
```suggestion
        List<Partition> partitions = Lists.newArrayList();
        if (!MapUtils.isEmpty(partVals)) {
         // Only load specify partitions
            CatalogClient client = new CatalogClient(source);
            partitions = client.getPartitions(srcDatabaseName, srcTableName, partVals);
         }
```